### PR TITLE
Allow to specify additional NBLS document selectors from other extensions.

### PR DESCRIPTION
**Pull request by dbalek**
Additional document selectors for NBLS can be specified in extension's `package.json` file as follows:
```
"contributes": {
	"netbeans.documentSelectors": [
		{
			"language": "yaml",
			"pattern": "**/{application,bootstrap}*.yml"
		},
		{
			"language": "properties",
			"pattern": "**/{application,bootstrap}*.properties"
		}
	]
	...
}
```

### CODE REVIEW

File: java/java.lsp.server/vscode/schemas/ext.package.schema.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "LSP Extension contributions to package.json",
+    "type": "object",
+    "properties": {
+        "contributes": {
+            "type": "object",
+            "properties": {
+                "nbls.documentSelectors" : {
```
Comment (sdedic) [inline java/java.lsp.server/vscode/schemas/ext.package.schema.json:9]: Q: the other (image resource map) uses `netbeans.` prefix. Shouldn't we use the same one ? If `nbls.` is now preferred (@jlahoda  ?), then please file an issue for me to fix the resource map setting.

Another question: is it recommended to add schema for each ferature ?

File: java/java.lsp.server/vscode/src/extension.ts [hunk 2]
```diff
@@ -385,11 +385,14 @@ export function activate(context: ExtensionContext): VSNetBeansAPI {
     // find acceptable JDK and launch the Java part
     findJDK((specifiedJDK) => {
         let currentClusters = findClusters(context.extensionPath).sort();
+        let currentDocumentSelectors = collectDocumentSelectors();
         context.subscriptions.push(vscode.extensions.onDidChange(() => {
             checkConflict();
             const newClusters = findClusters(context.extensionPath).sort();
-            if (newClusters.length !== currentClusters.length || newClusters.find((value, index) => value !== currentClusters[index])) {
+            const newDocumentSelectors = collectDocumentSelectors();
+            if (newClusters.length !== currentClusters.length || newDocumentSelectors.length !== currentDocumentSelectors.length || newClusters.find((value, index) => value !== currentClusters[index])) {
```
[the remark this answers is not in the document]
Comment (dbalek) [reply java/java.lsp.server/vscode/src/extension.ts]: Fixed.

File: java/java.lsp.server/vscode/schemas/ext.package.schema.json (hunk 1 above)
Comment (dbalek) [reply java/java.lsp.server/vscode/schemas/ext.package.schema.json]: Both features use `netbeans` prefix and are merged to the single schema now.

File: java/java.lsp.server/vscode/src/extension.ts [hunk 3]
```diff
@@ -1311,6 +1322,20 @@ export function deactivate(): Thenable<void> {
     return stopClient(client);
 }
 
+function collectDocumentSelectors(): TextDocumentFilter[] {
+    const selectors = [];
+    for (const extension of vscode.extensions.all) {
+        const contributesSection = extension.packageJSON['contributes'];
+        if (contributesSection) {
+            const documentSelectors = contributesSection['nbls.documentSelectors'];
```
Comment (sdedic) [inline java/java.lsp.server/vscode/src/extension.ts:1330]: reference to the previous `nbls.` prefix (schema was changed to `netbeans.`

Comment (dbalek) [reply java/java.lsp.server/vscode/src/extension.ts]: OOPS - fixed.

**Review by sdedic — APPROVED** [review]

**Review by jtulach — APPROVED** [review]

**Pull request merged**